### ISSUE TITLE
Inherit from `tk.StringVar` and overload `.set` method

### ISSUE THREAD

**Issue by WK-GiHu** [JDJacinto29/StackOverflow#7]
@JDJacinto29 

>the script now appears to run without crashing. 

OK, i would suggest to redefine `updateIP` and `updateSM` to a custom object to get the same behavior as using `.trace`

- [x] Yes
- [ ] No

_Originally posted by @WK-GiHu in https://github.com/JDJacinto29/StackOverflow/issues/6#issuecomment-574267286_

**Comment by WK-GiHu** [comment]
@JDJacinto29 

Insert before Line 75
https://github.com/JDJacinto29/StackOverflow/blob/8b4e85f8a74f0fbe75226a5a2ea964b7ed59b71f/SetIPexample.py#L75

```
def write_config():
    # save to config file
    with open(configFile, 'w') as fh:
        config.write(fh)
```
and
```
class UpdateIP(StringVar):
    def set(self, ip):
        print('UpdateIP.set({})'.format(ip))
        # update IP Address
        config[tkvar.get()]['TargetIP'] = ip
        write_config()

        super().set(ip)
```

Disable or remove this complete function definition, it's not longer used.
https://github.com/JDJacinto29/StackOverflow/blob/8b4e85f8a74f0fbe75226a5a2ea964b7ed59b71f/SetIPexample.py#L234

Change the Line 113
https://github.com/JDJacinto29/StackOverflow/blob/8b4e85f8a74f0fbe75226a5a2ea964b7ed59b71f/SetIPexample.py#L113

to 

```
updateIP = UpdateIP(root) 
```

**Comment by JDJacinto29** [comment]
@WK-GiHu Yeah that's working great, thanks! Will I still be able to live update the IP Address entered without using trace though? Previously I was using trace to tell when a user typed a change into the input field, and saving to the config file with each change

**Comment by WK-GiHu** [comment]
@JDJacinto29 

> Will I still be able to live update the IP Address entered without using trace though?

Please rephrase, i didn't get your sentence?

Using `.trace` will **update at every inputed character**, i couldn't recommend this.

**Comment by JDJacinto29** [comment]
@WK-GiHu Yea that's what I was looking for, to have it update on ever character. But it looks like right now its not updating the config file with new input at all. No problem, I should be able to take it from here and figure out what to do with that. Thank you fro your help!
